### DISCUSSION TITLE
How to set Schema property as unique

### DISCUSSION THREAD

**Discussion by ajunca** [ory/kratos#1185] [Q&A] [answered]
Hi,

I have defined a json schema with the main credentials as email + password. I also add a "username" text field so users can have a display name. My question is, how  can I make sure that the username is not already taken when a new registration is performed? I also want the username to be modifiable. 

What will be the best way to achieve something like this?

Thanks.

**Comment by zepatrik** [comment, marked as answer]
Currently Kratos does not support this. You have to move that logic to your application and update the username through the admin API if it is updated by the user.
This requires #249 and others.

**Reply by harnash** [reply]
$0.2 from me. Currently this is a quite problematic having username + password as credentials and have account recovery enabled (which is done via email). Two users can have two identical emails in the system and account recovery is being sent for the first account created which could be a bit surprising at least.

**Reply by aeneasr** [reply]
Recovery [emails should be unique](https://github.com/ory/kratos/blob/master/persistence/sql/migrations/templates/20200519101057_create_recovery_addresses.up.fizz#L11).

**Reply by harnash** [reply]
Ooops. It seems so indeed. Need to consult with the QA steps it was reproduced with. Could be something else or the problem might lie somewhere between keyboard and the screen ;-)

**Reply by aeneasr** [reply]
:)
